catalog/cataloger_merge.go

```diff
@@ -128,6 +125,43 @@ func (c *cataloger) Merge(ctx context.Context, repository, leftBranch, rightBran
 	return mergeResult, err
 }
 
+func drainChannel(channel chan mergeBatchRecords) {
```

Comment:
If you will communicate the diff scanner work throw 2 channels - data and error (size 1) - the optimization to drain the channel in all the places the scanner failed will be gone.
The merge/caller will check both channels and in case of an error, it will just exit.